spec/resources/api/v2/barcode_printer_resource_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require './app/resources/api/v2/barcode_printer_resource'
+
+RSpec.describe Api::V2::BarcodePrinterResource, type: :resource do
+  subject(:resource) { described_class.new(resource_model, {}) }
+
+  let(:resource_model) { build_stubbed :barcode_printer }
+
+  # Test attributes
+  it 'has the expected attributes', :aggregate_failures do
+    expect(resource).not_to have_attribute :id
+    expect(resource).to have_attribute :uuid
+    expect(resource).to have_attribute :name
+    expect(resource).to have_attribute :print_service
+    expect(resource).to have_attribute :barcode_type
+  end
+
+  # Updatable fields
+  it 'allows updating of read-write fields', :aggregate_failures do
+    expect(resource).to have_updatable_field :name
```

Comment:
If the resource is immutable is it actually possible to update any of these fields?

Reply:
No, but that's not managed at the resource level.  The request handler is the only thing that looks at the immutability of the resource itself.  I don't really like the way these tests have to be put together because of that.  I think the request test file does a might nicer job of describing the behaviour of the resource.

Reply:
Yeah these resources specs seem to be a repetition of alot of the requests test files. Traction doesn't test resources directly, only through the api.